src/app/containers/ConsentBanner/index.canonical.test.jsx

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable global-require */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { act } from 'react-dom/test-utils';
+
+let CanonicalContainer;
+let logic;
+let container;
+
+let setShowPrivacyBannerState;
+let setShowCookieBannerState;
+
+const runInitial = jest.fn();
+const privacyOnAllow = jest.fn();
+const privacyOnReject = jest.fn();
+const cookieOnAllow = jest.fn();
+const cookieOnReject = jest.fn();
+
+jest.mock('./Banner', () => jest.fn());
+const Banner = require('./Banner');
+
+Banner.mockImplementation(({ type }) => <div>Canonical {type} banner</div>);
+
+describe('Canonical Container', () => {
+  beforeEach(() => {
+    jest.mock('./CanonicalLogic', () => jest.fn());
+    logic = require('./CanonicalLogic');
+
+    logic.mockImplementation(
+      ({ setShowPrivacyBanner, setShowCookieBanner }) => {
+        setShowCookieBannerState = setShowCookieBanner;
+        setShowPrivacyBannerState = setShowPrivacyBanner;
+
+        return {
+          runInitial,
+          privacyOnAllow,
+          privacyOnReject,
+          cookieOnAllow,
+          cookieOnReject,
+        };
+      },
+    );
+
+    CanonicalContainer = require('./index').default;
+
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render privacy banner when both banners are set to be shown', () => {
+    act(() => {
```

Comment:
_This_ is where I agree with you 😄 about the `it`, `act`, `expect` cycle. Better this than random newlines!